src/RWXLoader.js

```diff
@@ -2014,21 +2024,29 @@ class RWXLoader extends Loader {
 			res = this.textureRegex.exec( line );
 			if ( this.enableTextures && res != null ) {
 
+				const textureExtension = res[ 3 ] != null ? res[ 3 ] : '.jpg';
 				const texture = res[ 2 ].toLowerCase();
-
 				if ( texture == 'null' ) {
 
 					ctx.materialTracker.currentRWXMaterial.texture = null;
 
 				} else {
 
-					ctx.materialTracker.currentRWXMaterial.texture = texture;
+					ctx.materialTracker.currentRWXMaterial.texture = texture + textureExtension;
 
 				}
 
 				if ( res[ 4 ] !== undefined ) {
 
-					ctx.materialTracker.currentRWXMaterial.mask = res[ 4 ];
+					if ( textureExtension === '.png' ) {
+
+						ctx.materialTracker.currentRWXMaterial.mask = res[ 4 ];
+
+					} else {
+
+						ctx.materialTracker.currentRWXMaterial.mask = res[ 5 ];
+
+					}
```

Comment:
I've tested various scenarios using the texture regex and I think you should just always pick `res[ 5 ]` (no need to test for any specific extension), try for yourself with and without extension on the texture name here: https://regex101.com/r/2wQnJa/1